src/main/java/com/bart/scorebetlive442/service/MatchService.java

```diff
@@ -1,37 +1,82 @@
 package com.bart.scorebetlive442.service;
 
+import com.bart.scorebetlive442.entity.MatchEntity;
+import com.bart.scorebetlive442.entity.TeamEntity;
+import com.bart.scorebetlive442.mapper.MatchMapper;
+import com.bart.scorebetlive442.mapper.TeamMapper;
 import com.bart.scorebetlive442.model.Match;
+import com.bart.scorebetlive442.repository.MatchRepository;
+import com.bart.scorebetlive442.repository.TeamRepository;
+import jakarta.persistence.EntityNotFoundException;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class MatchService {
 
-    private final List<Match> matchesList = new ArrayList<>();
-    private Long currentMatchId = 1L;
+    private final MatchRepository matchRepository;
+    private final MatchMapper matchMapper;
+    private final TeamMapper teamMapper;
+
+    public MatchService(MatchRepository matchRepository, MatchMapper matchMapper, TeamMapper teamMapper) {
+        this.matchRepository = matchRepository;
+        this.matchMapper = matchMapper;
+        this.teamMapper = teamMapper;
+
+    }
 
     public Match createMatch(Match match) {
-        match.setId(currentMatchId++);
-        matchesList.add(match);
-        return match;
+        MatchEntity toSave = matchMapper.toMatchEntity(match);
+        MatchEntity saved = matchRepository.save(toSave);
+
+        return matchMapper.toMatchModel(saved);
     }
 
     public Match getMatch(Long id) {
-        for (Match match : matchesList) {
-            if (match.getId().equals(id)) {
-                return match;
-            }
-        }
-        return null;
+      return matchRepository.findById(id)
+              .map(matchMapper::toMatchModel)
+              .orElse(null);
     }
 
     public List<Match> getAllMatches() {
-        return matchesList;
+        return matchRepository.findAll()
+                .stream()
+                .map(matchMapper::toMatchModel)
+                .collect(Collectors.toList());
+
     }
 
     public boolean removeMatchById(Long id) {
-        return matchesList.removeIf(match -> match.getId().equals(id));
+        if(matchRepository.existsById(id)) {
+            matchRepository.deleteById(id);
+            return true;
+        } else
+            return false;
+    }
+
+    public Match updateMatch(Long id, Match match) {
+        MatchEntity matchById = matchRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Match with id " + id + "not found"));
+
+        if(match.getCity() !=null) matchById.setCity(match.getCity());
```

Comment:
teoretycznie to też da się ograć mapstructem (bardziej zaawansowane): da się stworzyć metodę ktora przyjmuje:

MatchEntity updateME(MatchEntity current, Match received);

i tak ustawić mappery by przenosić z received tylko pola nienullowe do current

Reply:
postaraj się sam to rozgryźć za pomocą dokumentacji / chatgpt :)